var/spack/repos/builtin/packages/genemark-et/package.py

```diff
@@ -27,12 +27,17 @@ class GenemarkEt(Package):
     def url_for_version(self, version):
         return "file://{0}/gm_et_linux_64.tar.gz".format(os.getcwd())
 
+    def patch(self):
+        with working_dir(join_path(self.stage.source_path, "gmes_petap")):
+            perlscripts = glob.glob('*.pl')
+            filter_file('#!/usr/bin/perl', '#!/usr/bin/env perl', *perlscripts)
```

Comment:
(minor) I think you could also do 

```
filter_file('#!/usr/bin/perl', 
  '#!' + self.spec['perl'].prefix,
  *perlscripts)
```

This would allow running the installed scripts without prepending the `perl` bin directory to `PATH`

I would have sworn we did substitutions for `perl` shebangs but I couldn't find that; I also couldn't find it for python, although e.g. in the Spack Python package there is logic which rewrites the Spack Python prefix which adding Python to a view.